quinn-proto/src/connection/assembler.rs

```diff
@@ -383,4 +421,35 @@ mod test {
         assert_eq!(x.pop(), Some((0, Bytes::from_static(b"abcdef"))));
         assert_eq!(x.pop(), Some((9, Bytes::from_static(b"jklmno"))));
     }
+
+    #[test]
+    fn unordered_happy_path() {
+        let mut x = Assembler::new();
+        x.insert(0, Bytes::from_static(b"abc"));
+        assert_eq!(x.read_unordered(), Some((0, Bytes::from_static(b"abc"))));
+        assert_eq!(x.read_unordered(), None);
+        x.insert(3, Bytes::from_static(b"def"));
+        assert_eq!(x.read_unordered(), Some((3, Bytes::from_static(b"def"))));
+        assert_eq!(x.read_unordered(), None);
+    }
+
+    #[test]
+    fn unordered_dedup() {
```

Comment:
I think this is testing "middle" and "tail" duplication but not "head" duplication. Should we have a test for that as well to make sure we cover all the edge cases here?

Reply:
Added a few lines to exercise that too.